superset/utils/machine_auth.py

```diff
@@ -58,17 +66,52 @@ def authenticate_webdriver(
         # Setting cookies requires doing a request first
         driver.get(headless_url("/login/"))
 
+        cookies = self.get_cookies(user)
+
+        for cookie_name, cookie_val in cookies.items():
+            driver.add_cookie({"name": cookie_name, "value": cookie_val})
+
+        return driver
+
+    def authenticate_browser_context(
+        self,
+        browser_context: BrowserContext,
+        user: User,
+    ) -> BrowserContext:
+        # Short-circuit this method if we have an override configured
+        if self._auth_context_func_override:  # type: ignore
+            return self._auth_context_func_override(browser_context, user)
+
+        # Setting cookies requires doing a request first
+        page = browser_context.new_page()
+        page.goto(headless_url("/login/"))
+
+        cookies = self.get_cookies(user)
+
+        browser_context.clear_cookies()
+        browser_context.add_cookies(
+            [
+                {
+                    "name": cookie_name,
+                    "value": cookie_val,
+                    "domain": "0.0.0.0",
+                    "path": "/",
+                    "sameSite": "Lax",
+                    "httpOnly": True,
+                }
+                for cookie_name, cookie_val in cookies.items()
+            ]
+        )
```

Comment:
This didn't work for me in a setting where the baseurl isn't an ip address. However, with this change I was able to make it work. Looks slightly hacky, WDYT?

```python
        from urllib.parse import urlparse
        url = urlparse(current_app.config["WEBDRIVER_BASEURL"])

        browser_context.clear_cookies()
        browser_context.add_cookies(
            [
                {
                    "name": cookie_name,
                    "value": cookie_val,
                    "domain": url.netloc,
                    "path": "/",
                    "sameSite": "Lax",
                    "httpOnly": True,
                }
                for cookie_name, cookie_val in cookies.items()
            ]
        )
```

Reply:
Great point, hardcoding the domain was a big mistake!